### ISSUE TITLE
Add note(s) to docs about flat configs and/or no-unused-modules' `unusedExports` not being supported in flat configs

### ISSUE THREAD

**Issue by nnmrts** [import-js/eslint-plugin-import#2964]
I think for most users of this plugin it's not immediately obvious that flat configs are not or only partially supported. See for example the mentions of this issue in various commits, PRs and issues: https://github.com/import-js/eslint-plugin-import/issues/2556.

If I understand correctly after going through some of the issues here, the flat config format in general is not "officially" supported by this plugin, however it's not clear to me what exactly breaks and what not. I only found [this](https://github.com/import-js/eslint-plugin-import/pull/2873) (which aims to add support but doesn't have a lot of information) and [this comment here](https://github.com/import-js/eslint-plugin-import/pull/2829#issuecomment-1802847559):

> @keijokapp going from "we do not support flat config" to "we support it except for one rule which is horrible broken" isn't an improvement imo, since it will discourage people from using the rule in a way that will last long after it's eventually made to work.
> -- @ljharb 

In my opinion it should be made clear somewhere in the readme what "we do not support flat config" actually means. For me personally at least everything except for setting `unusedExports` to `true` in `no-unused-modules` "works" though.

Which brings me to the main point of this issue: Can we document this fact in https://github.com/import-js/eslint-plugin-import/blob/main/docs/rules/no-unused-modules.md? Especially because both the "In the future, these options will be enabled by default (see https://github.com/import-js/eslint-plugin-import/issues/1324)" and the example below might encourage readers to set `unusedExports` to `true`. Maybe a link to https://github.com/import-js/eslint-plugin-import/issues/2907 should be included.

**Comment by ljharb** [comment]
Given that eslint's FlatCompat allows people to use this plugin before it actually supports flat config, I agree adding a note in the readme (linked to from the no-unused-modules docs) here would be good.

**Comment by jpzwarte** [comment]
I'm not using the `no-usused-modules` rule directly, but I am attempting to bundle my ESLint flat config that includes `eslint-plugin-import` using rollup. 

Of all the plugins & rules I'm using, `no-unused-modules` is the only one that's giving me trouble:

```
(!) Plugin node-resolve: Could not resolve import "eslint/lib/cli-engine/file-enumerator" in /Users/jzwartepoorte/Projects/components/node_modules/eslint-plugin-import/lib/rules/no-unused-modules.js using exports defined in /Users/jzwartepoorte/Projects/components/node_modules/eslint/package.json.
(!) Plugin node-resolve: Could not resolve import "eslint/lib/util/glob-utils" in /Users/jzwartepoorte/Projects/components/node_modules/eslint-plugin-import/lib/rules/no-unused-modules.js using exports defined in /Users/jzwartepoorte/Projects/components/node_modules/eslint/package.json.
(!) Plugin node-resolve: Could not resolve import "eslint/lib/util/glob-util" in /Users/jzwartepoorte/Projects/components/node_modules/eslint-plugin-import/lib/rules/no-unused-modules.js using exports defined in /Users/jzwartepoorte/Projects/components/node_modules/eslint/package.json.
```

My next step is trying to exclude this rule somehow from bundling.

**Comment by ljharb** [comment]
I’m not sure why you’d ever want to bundle an eslint config - and either way, this plugin requires everything to exist on disk to work properly, so it’s not going to work with that approach.

**Comment by jpzwarte** [comment]
I'm trying to create a zero dependency ESLint config, which hopefully speeds up ESLint's running time significantly.

**Comment by ljharb** [comment]
That seems like a really bizarre way to solve it; I'd suggest using jest-eslint-runner instead, it does the job without violating the expectations of basically every package.

It seems like the error is coming from your rollup setup - the paths don't necessarily exist, which is why they're being conditionally required in the first place.

**Comment by lgenzelis** [comment]
For anyone else landing here looking for a solution, here's my `eslint.config.cjs`, which works for me :)

```js
const pluginImport = require('eslint-plugin-import');

module.exports = [
  {
    plugins: {
      import: { rules: pluginImport.rules },
    },
    rules: {
      'import/order': 'error',
      'import/group-exports': 'error',
      'import/exports-last': 'error',
    },
  },
  // ... all your other other configurations
];
```

**Comment by chrisdothtml** [comment]
@lgenzelis that doesn't solve the problem, as trying to use `import/no-unused-modules` still results in a "broken" config

**Comment by quememo** [comment]
is `missingExports` also not supported or only `unusedExports`?

**Comment by ronbraha** [comment]
I upgraded to the v2.31.0, which officially supports ESLint V9, enabling the `import/no-unused-modules` with the existing `unusedExports` option results in the ESLint CLI not recognizing the `eslint.config.js` unless there's a `.eslintrc` file next to it (the `.eslintrc` file can be empty, it just ignores it).

```
Oops! Something went wrong! :(

ESLint: 9.11.1

ESLint couldn't find a configuration file. To set up a configuration file for this project, please run:

    npm init @eslint/config@latest

ESLint looked for configuration files in <project rootDir> and its ancestors. If it found none, it then looked in your home directory.

If you think you already have a configuration file or if you need more help, please stop by the ESLint Discord server: https://eslint.org/chat
```

As long as the `.eslintrc` exists, the rule works as expected. Is that a known issue?

**Comment by ljharb** [comment]
In theory it’s now all supported, with v2.30 and v2.31. Please comment if not and I’ll reopen.

**Comment by shcallaway** [comment]
I'm on version 2.31 and I'm still getting a broken config with `unusedExports`.

From my lockfile:
```
      eslint-plugin-import:
        specifier: ^2.31.0
        version: 2.31.0(@typescript-eslint/parser@8.12.2(eslint@9.13.0)(typescript@5.6.3))(eslint@9.13.0)
```

My config:
```typescript
import eslint from "@eslint/js";
import tseslint from "typescript-eslint";
import importPlugin from "eslint-plugin-import";

export default [
  eslint.configs.recommended,
  ...tseslint.configs.recommended,
  importPlugin.flatConfigs.recommended,
  importPlugin.flatConfigs.typescript,
  {
    rules: {
      "no-console": "error",
      "import/no-unresolved": "off",
      "import/no-unused-modules": [
        "error",
        {
          missingExports: true,
          unusedExports: true,
        },
      ],
    },
  },
];

```
